client/js/store/modules/auth.js

```diff
@@ -107,5 +107,51 @@ export default {
         );
       });
     },
+    async requestStatus(context) {
+      if (!window) {
+        // Swallow error, this call is only supported client side
+        return;
+      }
+
+      const { location } = window;
+      const isLobby = location.hash.startsWith('#/lobby');
+      const isGame = location.hash.startsWith('#/game');
+      // We first need to check if this is a game route, if it is we can not auth the user or
+      // it will break the game until we add reconnect/subscribe logic
+      // By stopping here, Vue will allow the user to reconnect via the relogin dialog instead
+      if (isGame) {
+        return;
+      }
+
+      try {
+        const response = await fetch('/user/status', {
+          credentials: 'include',
+        });
+        const status = await response.json();
+        const { authenticated, username } = status;
+
+        // If the user is not authenticated, we're done here
+        if (!authenticated) {
+          return response;
+        }
+
+        // If the user is authenticated and has a username, add it to the store
+        if (username) {
+          context.commit('authSuccess', username);
+        }
+
+        // If this is a lobby, redirect the user to the game list so they don't have to
+        // log back in again
+        if (isLobby) {
+          location.href = '/#/';
+          return;
+        }
+
+        return response;
+      } catch (err) {
+        context.commit('clearAuth');
+        throw new Error(err);
```

Comment:
Silly nitpick, but any reason to wrap this in a new error vs throwing the OG?
```suggestion
        throw err;
```

Reply:
So I'm torn about this. I still don't know the best practices surrounding this, https://www.geeksforgeeks.org/difference-between-throw-errormsg-and-throw-new-errormsg/. (which is more about `Error()` new `new Error()`, but hear me out.


I'm not actually calling throw in the try part of the block, I'm relying on the calls themselves to fail for one reason or the other to get me in to the catch block, and then calling new Error manually.

I guess it depends on the way fetch is sending it. When you throw a new error it will wrap the message, something like this:
![image](https://user-images.githubusercontent.com/414475/187963503-2ec6f603-c849-4234-af70-9e5fdba010d9.png)

I used this code in the browser console:
```
(async () => {
    try {
        const response = await fetch('http://localhost:1234');
        if (!response.ok) {
            throw new Error('response not ok');
        }
        return response;
    } catch (err) {
        console.log('err (before):', err);
        const newErr = new Error(err);
        console.log('err (after):', newErr);
        throw newErr;
    };
})();
```

All this leads back to "I don't know" so I did it the way I usually do.

**This did make me take a more top down view as to what I was trying to accomplish and since I was just catching the error in the `main.js` file and silently failing made me realize we don't need to throw an error at all so I removed it.**